bb8/src/internals.rs

```diff
@@ -15,6 +15,7 @@ where
     M: ManageConnection + Send,
 {
     pub(crate) statics: Builder<M>,
+    pub(crate) executor: Box<dyn Executor>,
```

Comment:
Why does this need a trait object? It would be nice if we didn't have to `Box` every future just to support this fairly niche use case.